apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/Operation.java

```diff
@@ -0,0 +1,137 @@
+package gov.cms.bfd.server.war;
+
+import com.codahale.metrics.servlets.HealthCheckServlet;
+import com.codahale.metrics.servlets.MetricsServlet;
+import com.codahale.metrics.servlets.PingServlet;
+import com.codahale.metrics.servlets.ThreadDumpServlet;
+import com.newrelic.api.agent.NewRelic;
+import gov.cms.bfd.server.war.stu3.providers.CoverageResourceProvider;
+import gov.cms.bfd.server.war.stu3.providers.ExplanationOfBenefitResourceProvider;
+import gov.cms.bfd.server.war.stu3.providers.PatientResourceProvider;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import javax.servlet.http.HttpServletRequest;
+import org.hl7.fhir.dstu3.hapi.rest.server.ServerCapabilityStatementProvider;
+import org.slf4j.MDC;
+
+/**
+ * Models the canonical "operations" supported by this application, such that each meaningfully
+ * distinct operation can be assigned a unique canonical name, for use in monitoring systems.
+ *
+ * <p>Also handles the publishing of those canonical names to the places they need to be published,
+ * via the {@link #publishOperationName()} method.
+ */
+public final class Operation {
```

Comment:
I like this class. I suggest that it may be improved by defining common options like "id". To avoid inconsistency for in calls to setOption("id", ...) or setOption("Id", ...) etc. This thinking is similar to the reason for an Endpoint enumeration.

Reply:
Thought about doing that but decided that, if I wanted to go there, it'd be better to create types or subtypes for every operation. Decided that wasn't worth the effort right now.